### PR TITLE
[super-agent] feat: add license key as env from secret (NR-293501)

### PR DESCRIPTION
**Pull request by gsanchezgavier**
#### What this PR does / why we need it:

- Adds license key as env variable  from the secret for the SA deployment. This will allow to use env vars placeholders in configs rather than plain secrets for sub-agents.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/super-agent/values.yaml [hunk 1]
```diff
@@ -138,7 +138,7 @@ super-agent-deployment:
         #   endpoint: https://opamp.service.newrelic.com/v1/opamp
         #   headers:
         #     # TODO this value should be automatically injected
-        #     api-key: LICENSE_KEY
+        #     api-key: ${nr-env:NR_LICENSE_KEY}
```
Comment (paologallinaharbur) [inline charts/super-agent/values.yaml:141]: I think this is wrong

File: charts/super-agent/charts/super-agent-deployment/templates/deployment-superagent.yaml [hunk 2]
```diff
@@ -54,8 +54,13 @@ spec:
...
           imagePullPolicy: {{ .Values.image.pullPolicy }}
 
-          {{- with .Values.extraEnv }}
           env:
+            - name: NR_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "newrelic.common.license.secretName" . }}
+                  key: {{ include "newrelic.common.license.secretKeyName" . }}
+          {{- with .Values.extraEnv }}
```
Comment (paologallinaharbur) [inline charts/super-agent/charts/super-agent-deployment/templates/deployment-superagent.yaml:63]: I would add an extra env var NR_SA_OPAMP__HEADERS__API-KEY so that we van remove the "opamp:" entry in the sa config

File: charts/super-agent/values.yaml (hunk 1 above)
Comment (paologallinaharbur) [reply charts/super-agent/values.yaml:141]: nice point 😕 I guess that also Juanma PR has the same issue of "enabling by default" opamp

Comment (paologallinaharbur) [reply charts/super-agent/values.yaml:141]: I am checking how complex it is to include enable/disable for opamp

File: charts/super-agent/charts/super-agent-deployment/tests/deployment_superagent_securityContext_test.yaml [hunk 3]
```diff
@@ -3,6 +3,9 @@ templates:
   - templates/deployment-superagent.yaml
   - templates/configmap-superagent-config.yaml
   - templates/configmap-subagent-configs.yaml
+release:
```
Comment (paologallinaharbur) [inline charts/super-agent/charts/super-agent-deployment/tests/deployment_superagent_securityContext_test.yaml:6]: why this change is needed?

**Review by paologallinaharbur — CHANGES_REQUESTED** [review]
All looks good, we need a release and to bump the image.

I would consider changing the cluster as well in a follow up PR if it is a few line change

File: charts/super-agent/charts/super-agent-deployment/tests/deployment_superagent_securityContext_test.yaml (hunk 3 above)
Comment (gsanchezgavier) [reply charts/super-agent/charts/super-agent-deployment/tests/deployment_superagent_securityContext_test.yaml:6]: leftover, since i ended up moving the test to other file

File: charts/super-agent/charts/super-agent-deployment/templates/deployment-superagent.yaml (hunk 2 above)
Comment (gsanchezgavier) [reply charts/super-agent/charts/super-agent-deployment/templates/deployment-superagent.yaml]: done

**Review by paologallinaharbur — APPROVED** [review]
